core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/security/CarbonJMXAuthenticator.java

```diff
@@ -77,7 +77,7 @@ public Subject authenticate(Object credentials) {
         // Perform authentication
         //
         String userName = aCredentials[0];
-        String password = aCredentials[1];
+        char[] password = aCredentials[1].toCharArray();
```

Comment:
password is used in the authenticate method https://github.com/wso2/carbon-kernel/pull/4107/files#diff-37af262b12876d6f47e8a7e0c71d5d5af2239502d1e035cb2ed375d31e56fbbfR107, However as authenticate method accept Object type for password, no need to convert it to a String again